src/monaco.contribution.ts

```diff
@@ -121,6 +121,23 @@ const javascriptDefaults = new LanguageServiceDefaultsImpl(
 	{ allowNonTsExtensions: true, allowJs: true, target: ScriptTarget.Latest },
 	{ noSemanticValidation: true, noSyntaxValidation: false });
 
+function getTypeScriptWorker(): monaco.Promise<Worker> {
+	return new monaco.Promise((resolve, reject) => {
+ 		withMode((mode) => {
+ 			mode.getTypeScriptWorker()
+ 				.then(resolve, reject);
+ 		});
+ 	});
+}
+
+function getJavaScriptWorker(): monaco.Promise<Worker> {
+	return new monaco.Promise((resolve, reject) => {
+ 		withMode((mode) => {
+ 			mode.getJavaScriptWorker()
+ 				.then(resolve, reject);
+ 		});
+ 	});

```

Comment:
why so complex and not just like `return withMode(mode => mode.getJavaScriptWorker())`?